tensorflow/python/ops/image_ops_test.py

```diff
@@ -6047,12 +6047,31 @@ def testBroadcast(self):
     img = constant_op.constant(np.concatenate(img))
     img1 = array_ops.expand_dims(img, axis=0)  # batch dims: 1, 2.
     img2 = array_ops.expand_dims(img, axis=1)  # batch dims: 2, 1.
-
     score_tensor = image_ops.ssim_multiscale(
         img1, img2, 1.0, filter_size=11, filter_sigma=1.5, k1=0.01, k2=0.03)
     with self.cached_session():
       self.assertAllClose(expected, self.evaluate(score_tensor), 1e-4)
 
+  def testRaiseWhenDimensionPowerFactorsAndFilterSizeNotCompatible(self):
+    """Test case for GitHub issue 33840.
+
+    MS-SSIM does downsampling by the same number of power_factors and 
+    filter_size should be adjusted by the smallest image size.
+    https://github.com/tensorflow/tensorflow/issues/33840#issuecomment-633715778
+    """
+    #  For filter_size 11 do you need an image at least 176X176
+    img1 = self._RandomImage((1, 150, 240, 3), 255)
+    img2 = self._RandomImage((1, 150, 240, 3), 255)
+    img1 = constant_op.constant(img1, dtypes.uint8)
+    img2 = constant_op.constant(img2, dtypes.uint8)
+    with self.cached_session():
+      with self.assertRaisesRegexp(
+        RuntimeError, 'You defined 5 "power_factors"'):
+        ssim_uint8 = image_ops.ssim_multiscale(
+          img1, img2, 255, filter_size=11, filter_sigma=1.5, k1=0.01, k2=0.03
+        )
+        self.evaluate(ssim_uint8)
```

Comment:
I'm getting an `InvalidArgumentError` here:
```
InvalidArgumentError: assertion failed: [9.375] [11]
```

Reply:
Can you please provide me more details? Because it's working in CI... Are you forcing graph mode for tests?

Reply:
Thank you for your help and debug! I'm new here and it's my first PR. I did this command here with my branch updated and getting success, but I'm trying to reproduce `InvalidArgumentError`. See what I got with the command:

![Captura de tela de 2023-08-30 09-20-23](https://github.com/tensorflow/tensorflow/assets/1108613/03695707-53fb-46c5-bc0b-db7b9167ab71)

Reply:
I did a change including InvalidArgumentError as a kind of error type similar to other tests

Reply:
This is still failing for the same error:
```
tensorflow.python.framework.errors_impl.InvalidArgumentError: assertion failed: [9.375] [11]
	 [[{{node MS-SSIM/Assert_2/AssertGuard/Assert}}]]
```

Reply:
now i catch error from right import